src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java

```diff
@@ -215,102 +345,118 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
-        analysisService.setVisibilityFilter(getVisibilityFilter());
+        analysisService.setVisibilityFilter(getVisibilityFilterOrDefault());
+        analysisService.setVisibilityFilterNewWarnings(getVisibilityFilterNewWarningsOrDefault());
 
+        String analysisId = null;
         if(StringUtils.isBlank(aid)) {
-            LOGGER.log(Level.INFO, "[CodeSonar] Determining analysis id...");
-            aid = workspace.act(new DetermineAid());
+            LOGGER.log(Level.INFO, "Determining analysis id...");
+            analysisId = workspace.act(new DetermineAid(expandedCodesonarProjectFile));
+            LOGGER.log(Level.INFO, "Found analysis id: {0}", analysisId);
         } else {
-            LOGGER.log(Level.INFO, "[CodeSonar] Using override analysis id: '" + aid + "'.");
+            analysisId = aid;
+            LOGGER.log(Level.INFO, "Using override analysis id: \"" + aid + "\".");
         }
         
-        String analysisUrl = baseHubUri.toString() + "/analysis/" + aid + ".xml";
+        String analysisUrl = baseHubUri.toString() + "/analysis/" + analysisId + ".xml";
 
         Analysis analysisWarnings = analysisService.getAnalysisFromUrlWarningsByFilter(analysisUrl);
-        URI metricsUri = metricsService.getMetricsUriFromAnAnalysisId(baseHubUri, aid);
+        URI metricsUri = metricsService.getMetricsUriFromAnAnalysisId(baseHubUri, analysisId);
         Metrics metrics = metricsService.getMetricsFromUri(metricsUri);
-        URI proceduresUri = proceduresService.getProceduresUriFromAnAnalysisId(baseHubUri, aid);
+        URI proceduresUri = proceduresService.getProceduresUriFromAnAnalysisId(baseHubUri, analysisId);
         Procedures procedures = proceduresService.getProceduresFromUri(proceduresUri);
 
         Analysis analysisNewWarnings = analysisService.getAnalysisFromUrlWithNewWarnings(analysisUrl);
         List<Pair<String, String>> conditionNamesAndResults = new ArrayList<>();
-
-        CodeSonarBuildActionDTO buildActionDTO = new CodeSonarBuildActionDTO(analysisWarnings, analysisNewWarnings, metrics, procedures, baseHubUri);
+        
+        Long lAnalysisId = null;
+        try {
+            lAnalysisId = Long.valueOf(analysisId);
+        } catch(NumberFormatException e) {
+            LOGGER.log(Level.WARNING, "Unable to parse analysis id \"" + analysisId + "\" as long integer.");
+        }
+        CodeSonarBuildActionDTO buildActionDTO = new CodeSonarBuildActionDTO(lAnalysisId, analysisWarnings, analysisNewWarnings, metrics, procedures, baseHubUri);
         CodeSonarBuildAction csba = new CodeSonarBuildAction(buildActionDTO, run, expandedProjectName, analysisUrl);
         
-        listener.getLogger().println("[CodeSonar] Finding previous builds for comparison");
+        csLogger.writeInfo("Finding previous builds for comparison");
         
         CodeSonarBuildActionDTO compareDTO = null;
         Run<?,?> previosSuccess = run.getPreviousSuccessfulBuild();
```

Comment:
I think it is about time we fix the spelling of this variable: "previosSuccess" -> "previousSuccess"